### PR TITLE
GitIgnore add wwwroot, node_modules, sln

### PR DESCRIPTION
**Pull request by Rick-Anderson**

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@spboyer @tdykstra  please review

**Comment by bradygaster** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.gitignore](https://github.com/aspnet/Docs/blob/Rick-Anderson-gitignore/.gitignore) | :white_check_mark:Succeeded | | [Details](#user-content-.gitignore)


For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C18%5C6a0d29ae-bb37-09f1-c55a-4cfa862c6784%5CPullRequest%5C201611180040329900-2203%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,13 @@
 _build/
 _site/
+Properties/
+node_modules/
+wwwroot/
+
+#override .sln when required. Not needed in most cases.
+*.sln
+
+Project_Readme.html
```
Comment (tdykstra) [inline .gitignore]: I would rather not include .sln here.  There are times when we need it, and it's simple enough to omit it if we don't need it, and it doesn't hurt anything if it's there when we don't need it.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,13 @@
 _build/
 _site/
+Properties/
+node_modules/
+wwwroot/
+
+#override .sln when required. Not needed in most cases.
+*.sln
```
Comment (tdykstra) [inline .gitignore]: I prefer not to include .sln here.  There are times when you need it, and overriding a .gitignore is going to be a nuisance, and it doesn't hurt anything if it's there when you don't need it.

File: .gitignore (hunk 1 above)
Comment (tdykstra) [inline .gitignore]: Better note would be good, but before we go that route I'd like to get Shayne's take on it, as I'm still not convinced it's the best choice.

**Comment by Rick-Anderson** [comment]
@spboyer  title says everything :) so does file diff

**Comment by Rick-Anderson** [comment]
@tdykstra  add command to force add of .sln file
The PU is requesting we check in min files needed.

**Comment by bradygaster** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.gitignore](https://github.com/aspnet/Docs/blob/Rick-Anderson-gitignore/.gitignore) | :white_check_mark:Succeeded | | [Details](#user-content-.gitignore)


For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C18%5C6a0d29ae-bb37-09f1-c55a-4cfa862c6784%5CPullRequest%5C201611180103221732-2203%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by tdykstra** [comment]
Checking in minimum number of files is a goal, not a way of attaining the goal.  I'm not convinced this is a good way to attain that goal.

**Comment by Rick-Anderson** [comment]
@danroth27  and I talked about this. It's easy to force add and you only need to do that once. That's the exception, most projects don't require it. When you allow .sln, then ever commit you need to manually remove it. Right now we have 35 of them, most are not needed. It's far less work and safer to ignore by default.

**Comment by Rick-Anderson** [comment]
@tdykstra I asked @ardalis  to manually remove .sln and he pointed out VS auto creates this file. Each time you commit you're then forced to manually remove it. Given it's trivial to force add, and once force added there is never any manual work.

That's what .gitignore is for, default rule so we don't have to police this.

**Comment by bradygaster** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[.gitignore](https://github.com/aspnet/Docs/blob/Rick-Anderson-gitignore/.gitignore) | :white_check_mark:Succeeded | | [Details](#user-content-.gitignore)


For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2016%5C11%5C18%5C6a0d29ae-bb37-09f1-c55a-4cfa862c6784%5CPullRequest%5C201611180119531166-2203%5Cworkflow_report.html).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**